packages/yoastseo/spec/parse/build/private/getTextElementPositionsSpec.js

```diff
@@ -26,249 +29,220 @@ describe( "A test for getting positions of sentences", () => {
+		expect( comma2.sourceCodeRange ).toEqual( { startOffset: 28, endOffset: 29 } );
+		expect( space3.sourceCodeRange ).toEqual( { startOffset: 29, endOffset: 30 } );
+		expect( yoast.sourceCodeRange ).toEqual( { startOffset: 34, endOffset: 39 } );
+		expect( bang2.sourceCodeRange ).toEqual( { startOffset: 39, endOffset: 40 } );
 
 		const sentences = [ { text: "Hello, world!" }, { text: " Hello, yoast!" } ];
-		const sentencesWithPositions = [ { text: "Hello, world!", sourceCodeRange: { startOffset: 8, endOffset: 21 } },
-			{ text: " Hello, yoast!", sourceCodeRange: { startOffset: 21, endOffset: 57 } } ];
-
-		expect( getTextElementPositions( node, sentences ) ).toEqual( sentencesWithPositions );
+		const [ helloSentence, yoastSentence ] = getTextElementPositions( paragraph, sentences );
+		expect( helloSentence.sourceCodeRange ).toEqual( { startOffset: 3, endOffset: 16 } );
+		expect( yoastSentence.sourceCodeRange ).toEqual( { startOffset: 22, endOffset: 40 } );
 	} );
 
 	it( "gets the sentence positions from an implicit paragraph", function() {
 		// HTML: <div>Hello <em>World!</em></div>.
-		const node = new Paragraph( {}, [ { name: "#text", value: "Hello, World!" },
-			{
-				name: "em",
-				attributes: {},
-				childNodes: [ {
-					name: "#text",
-					value: "World!",
-				} ],
-				sourceCodeLocation: {
-					startOffset: 11,
-					endOffset: 26,
-					startTag: {
-						startOffset: 11,
-						endOffset: 15,
-					},
-					endTag: {
-						startOffset: 21,
-						endOffset: 26,
-					},
-				},
-			} ],
-		{
-			startOffset: 5,
-			endOffset: 32,
-		},
-		true );
+
+		const html = "<div>Hello <em>World!</em></div>";
+		const tree = adapt( parseFragment( html, { sourceCodeLocationInfo: true } ) );
+		const paragraph = tree.childNodes[ 0 ];
+		const tokens = [ "Hello", " ", "World", "!" ].map( string => new Token( string ) );
+
+		const [ hello, space, world, bang ] = getTextElementPositions( paragraph, tokens );
+
+		expect( hello.sourceCodeRange ).toEqual( { startOffset: 5, endOffset: 10 } );
+		expect( space.sourceCodeRange ).toEqual( { startOffset: 10, endOffset: 11 } );
+		expect( world.sourceCodeRange ).toEqual( { startOffset: 15, endOffset: 20 } );
+		expect( bang.sourceCodeRange ).toEqual( { startOffset: 20, endOffset: 21 } );
 
 		const sentences = [ { text: "Hello World!" } ];
-		const sentencesWithPositions = [ { text: "Hello World!", sourceCodeRange: { startOffset: 5, endOffset: 26 } } ];
+		const [ helloSentence ] = getTextElementPositions( paragraph, sentences );
+		expect( helloSentence.sourceCodeRange ).toEqual( { startOffset: 5, endOffset: 21 } );
+	} );
 
-		expect( getTextElementPositions( node, sentences ) ).toEqual( sentencesWithPositions );
+	it.skip( "should get the correct sentence position for a sentence in an image caption", function() {
+		// html: "<p><img class='size-medium wp-image-33' src='http://basic.wordpress.test/wp-content/uploads/2021/08/" +
```

Comment:
It can be removed. But I put it there to keep consistent with the rest of the file. Each spec starts with a comment containing a readable version of the html. So I sticked with that. I did edit the comment as to give it more value.